core/block/source/source_test.go

```diff
@@ -42,3 +56,108 @@ func Test_snapshotChance2(t *testing.T) {
 	// here is an example of distribution histogram
 	// https://docs.google.com/spreadsheets/d/1xgH7fUxno5Rm-0VEaSD4LsTHeGeUXQFmHsOm29M6paI
 }
+
+func TestSource_PushChange(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	initTreeAndSource := func() (ot *mock_objecttree.MockObjectTree, s *source) {
+		ot = mock_objecttree.NewMockObjectTree(ctrl)
+		ot.EXPECT().Root().Return(&objecttree.Change{Id: ""})
+		ot.EXPECT().IterateFrom(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil)
+		s = &source{ObjectTree: ot}
+		return
+	}
+
+	t.Run("big change", func(t *testing.T) {
+		//given
+		_, s := initTreeAndSource()
+		params := PushChangeParams{
+			State: state.NewDoc("", nil).NewState(),
+			Changes: []*pb.ChangeContent{{&pb.ChangeContentValueOfRelationAdd{RelationAdd: &pb.ChangeRelationAdd{
+				RelationLinks: []*model.RelationLink{{Key: strings.Repeat("r", changeSizeLimit)}}},
```

Comment:
So heavy tests. Extract size checking into separate function with maxSize parameter and you don't need to allocate so much memory

Reply:
Done!